R/Model-class.R

```diff
@@ -127,144 +127,130 @@ NULL
-##'
-##' @slot mean the prior mean vector \eqn{\mu}
-##' @slot cov the prior covariance matrix \eqn{\Sigma}
-##' @slot refDose the reference dose \eqn{x^{*}}
-##'
-##' @example examples/Model-class-LogisticLogNormal.R
-##' @export
-##' @keywords classes
-.LogisticLogNormal <-
-    setClass(Class="LogisticLogNormal",
-             representation(mean="numeric",
-                            cov="matrix",
-                            refDose="numeric"),
-             prototype(mean=c(0, 1),
-                       cov=diag(2),
-                       refDose=1),
-             contains="Model",
-             validity=
-                 function(object){
-                     o <- Validate()
+# LogisticLogNormal-class ----
 
-                     o$check(length(object@mean) == 2,
-                             "mean must have length 2")
-                     o$check(identical(dim(object@cov), c(2L, 2L)) &&
-                                 ! is.null(chol(object@cov)),
-                             "cov must be positive-definite 2x2 covariance matrix")
-                     o$check(is.scalar(object@refDose) &&
-                                 (object@refDose > 0),
-                             "refDose must be positive scalar")
-
-                     o$result()
-                 })
+#' `LogisticLogNormal`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' [`LogisticLogNormal`] is the class for a usual logistic regression model with
+#' a bivariate normal prior on the intercept and log slope.
+#'
+#' @details The covariate is the natural logarithm of the dose \eqn{x} divided
+#'   by the reference dose \eqn{x^{*}}:
+#'   \deqn{logit[p(x)] = \alpha + \beta \cdot \log(x/x^{*})},
+#'   where \eqn{p(x)} is the probability of observing a DLT for a given dose
+#'   \eqn{x}.
+#'   The prior is: \deqn{(\alpha, \log(\beta)) \sim Normal(\mu, \Sigma)}.
+#'   The slots of this class contain the mean vector and the covariance matrix
+#'   of the bivariate normal distribution, as well as the reference dose.
+#'
+#' @note Te parametrization inside the class uses `alpha0` and `alpha1`.
+#'   `alpha0` is identical to the intercept \eqn{\alpha} above and is the
+#'   log-odds for a DLT at the reference dose `x`*. Therefore, the prior mean
+#'   for `alpha0` is the expected log-odds at the reference dose `x`* before
+#'   observing any data. Note that the expected odds is not just the exp of the
+#'   prior mean of `alpha0`, because the non-linearity of the exp
+#'   transformation. The log-normal distribution on Wikipedia gives the formula
+#'   for computing the prior mean of `exp(alpha0)`, where `alpha0` is the
+#'   `log(alpha)` in the Neuenschwander et al. (2008) paper. The `alpha1` is
+#'   identical to \eqn{\beta} above and equals to `beta` in the Neuenschwander
+#'   et al paper. The `exp(alpha1)` gives the odds-ratio for DLT between two
+#'   doses that differ by the factor `exp(1)` ~ 2.7. Paramter `alpha1` has a
```

Comment:
```suggestion
#'   doses that differ by the factor `exp(1)` ~ 2.7. Parameter `alpha1` has a
```

Reply:
can you double check the wordlist file that it does not contain wrong spelling terms? thx

Reply:
Indeed, Tx @danielinteractive , The`WORDLIST` contains wrong `paramter`. I will correct it now!